MaterialLib/MPL/Utils/GetSymmetricTensor.cpp

```diff
@@ -0,0 +1,99 @@
+/*
+ * \file
+ * \copyright
+ * Copyright (c) 2012-2020, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ * Created on March 04, 2020, 5:20 PM
+ */
+
+#include "GetSymmetricTensor.h"
+
+#include "MaterialLib/MPL/PropertyType.h"
+
+namespace MaterialPropertyLib
+{
+template <int GlobalDim>
+struct GetSymmetricTensor
+{
+    SymmetricTensor<GlobalDim> operator()(double const& value) const
+    {
+        SymmetricTensor<GlobalDim> result = SymmetricTensor<GlobalDim>::Zero();
+        result.template head<3>() = Eigen::Vector3d::Constant(value);
+        return result;
+    }
+
+    SymmetricTensor<GlobalDim> operator()(Eigen::Vector2d const& values) const
+    {
+        SymmetricTensor<GlobalDim> result = SymmetricTensor<GlobalDim>::Zero();
+        result.template head<2>() = values;
+        return result;
+    }
+
+    SymmetricTensor<GlobalDim> operator()(Eigen::Vector3d const& values) const
+    {
+        SymmetricTensor<GlobalDim> result = SymmetricTensor<GlobalDim>::Zero();
+        result.template head<3>() = values;
+        return result;
+    }
+
+    SymmetricTensor<GlobalDim> operator()(Eigen::Matrix2d const& values) const
+    {
+        if constexpr (GlobalDim == 2)
+        {
+            SymmetricTensor<GlobalDim> result;
+            result << values(0, 0), values(1, 1), 0., values(0, 1);
+            return result;
+        }
+        OGS_FATAL("Cannot convert 2d matrix to 3d symmetric Tensor.");
```

Comment:
Can you try this instead (and actually call a wrong version somewhere):
```suggestion
    else {
        static_assert(GlobalDim == 2, "Cannot convert 2d matrix to 3d symmetric Tensor.");
    }
```
I think a compile-time error is possible here.

Reply:
It doesn't seem to work. The error triggers only when the expr is false and since in this case GlobalDim has to be 2 it is always true, so no error is thrown.
But further, if I static_assert GlobalDim == 3 it will always fail to compile, since it compiles this function for GlobalDim = 2 and for 3 anyway, even if I don't call the wrong function.